include/boost/hana/fwd/index_if.hpp

```diff
@@ -0,0 +1,63 @@
+/*!
+@file
+Forward declares `boost::hana::index_if`.
+
+@copyright Louis Dionne 2013-2017
+@copyright Jason Rice 2017
+Distributed under the Boost Software License, Version 1.0.
+(See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_HANA_FWD_INDEX_IF_HPP
+#define BOOST_HANA_FWD_INDEX_IF_HPP
+
+#include <boost/hana/config.hpp>
+#include <boost/hana/core/when.hpp>
+
+
+BOOST_HANA_NAMESPACE_BEGIN
+    //! Finds the value associated to the first key satisfying a predicate.
+    //! @ingroup group-Iterable
+    //!
+    //! Given an `Iterable` structure `xs` and a predicate `pred`,
+    //! `index_if(xs, pred)` returns an `IntegralConstant` index *of* the first
+    //! element whose key satisfies the predicate. If no element satisfies the
```

Comment:
Iterables have no notion of `key`, so just saying "index of the first element that satisfies the predicate" is enough.